### PR TITLE
Allow additional attributes from torch modules into SyMPC modules

### PR DESCRIPTION
**Pull request by kamathhrishi**
## Description
Currently, only weights and bias have been copied from torch modules to sympc modules (conv2d and Linear). But, additional attributes as in case of conv2d such as padding and stride haven't been copied. This results in differing functionality from intended behavior.

## How has this been tested?
- Modified test to include stride 

**Reproducing Bug** 
This could be reproduced in the current version of the codebase by modifying https://github.com/OpenMined/SyMPC/blob/090c11dbefbebaab160c6480ade2d3df120715a1/tests/sympc/module/module_test.py#L37 to include a stride, padding or any value that is different from the default (say 2). Further, modifying the linear layer dimensions accordingly. You will notice that test fails because behaviour deviates from intended behaviour (as done by torch on a given input). 

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: tests/sympc/module/module_test.py [hunk 1]
```diff
@@ -34,15 +34,15 @@ class ConvNet(sy.Module):
     def __init__(self, torch_ref, kernel_size=5):
```
Comment (gmuraru) [inline tests/sympc/module/module_test.py:34]: Q: Could you validate that stride was added to the module?

Comment (kamathhrishi) [reply tests/sympc/module/module_test.py:34]: Done.

File: src/sympc/module/nn/conv.py [hunk 2]
```diff
@@ -78,46 +80,65 @@ def forward(self, x: MPCTensor) -> MPCTensor:
...
         # we have groups == 1
+
         (
             self.out_channels,
             self.in_channels,
             kernel_size_w,
             kernel_size_h,
         ) = shape
 
-        if kernel_size_w != kernel_size_h:
```
Comment (kamathhrishi) [inline src/sympc/module/nn/conv.py:114]: Allow rectangular kernels.

File: src/sympc/module/nn/functional.py [hunk 3]
```diff
@@ -262,6 +262,9 @@ def max_pool2d(
     Raises:
         ValueError: if the kernel size is bigger than the input
     """
+    if x.session.nr_parties != 2:
```
Comment (kamathhrishi) [inline src/sympc/module/nn/functional.py:265]: Added check to ensure only 2 parties are present in session.

**Review by gmuraru — APPROVED** [review]
LGTM!

**Pull request merged**